### PR TITLE
ARROW-3264: [Java] Checkstyle fix whitespace

### PR DESCRIPTION
**Pull request by BryanCutler**
Enabled remaining whitespace checks
```
OperatorWrapCheck
SeparatorWrapCheck
NewlineAtEndOfFileCheck
WhitespaceAroundCheck
EmptyLineSeparatorCheck
```
Changed style operator wrap to be at end of line

### CODE REVIEW

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 1]
```diff
@@ -109,8 +109,8 @@ public ArrowBuf reallocIfNeeded(final int size) {
     if (bufManager != null) {
       return bufManager.replace(this, size);
     } else {
-      throw new UnsupportedOperationException("Realloc is only available in the context of an " +
-          "operator's UDFs");
+      throw new UnsupportedOperationException("Realloc is only available in the context of an "
+          + "operator's UDFs");
```
Comment (BryanCutler) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: Our current checkstyle rule for `OperatorWrapCheck` wants the operator on a new line like this. Doesn't matter to me, but I am more used to having it at the end of the line, which seems to be how it mostly is already. Thoughts @jacques-n or @icexelloss ?

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 2]
```diff
@@ -58,8 +58,8 @@
   private final BufferManager bufManager;
   private final ArrowByteBufAllocator alloc;
   private final boolean isEmpty;
-  private final HistoricalLog historicalLog = BaseAllocator.DEBUG ?
-      new HistoricalLog(BaseAllocator.DEBUG_LOG_LENGTH, "ArrowBuf[%d]", id) : null;
+  private final HistoricalLog historicalLog = BaseAllocator.DEBUG
+      ? new HistoricalLog(BaseAllocator.DEBUG_LOG_LENGTH, "ArrowBuf[%d]", id) : null;
```
Comment (icexelloss) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: Does the style require "?" to be on the second line?

Comment (BryanCutler) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: yup, see the above comment

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java (hunk 1 above)
Comment (icexelloss) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: I am used to having it at end of line as well. So light +1 on that.

Comment (BryanCutler) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: slight +1 for me too, so I will proceed with having operators on the end of the line

**Comment by BryanCutler** [comment]
@icexelloss @jacques-n this is ready, please take a final look, thanks!

**Comment by wesm** [comment]
This looks good to me, others?

**Pull request closed without merging**